### ISSUE TITLE
[Request] "Remove all torches" plugin

### ISSUE THREAD

**Issue by GregariousJB** [TEdit/Terraria-Map-Editor#1679]
Similar to #1273, though I thought it might be nice to have a quick 'n dirty option for those of us discovering that [biome torches now increase luck](https://terraria.wiki.gg/wiki/Luck#Torches).

Thought it would fit nicely beside the "remove all tiles" and "remove all chests" plugins and be fairly simple to implement in comparison to the sprite find/replace tool currently being worked on.

It would remove every single torch in the game, no matter what variation it is. Then, [Torch God's Favor](https://terraria.wiki.gg/wiki/Torch_God%27s_Favor) in hand, we could run back through our maps and quickly place torches using the smart cursor. This is also especially quick with [Journey mode's increased placement range](https://terraria.wiki.gg/wiki/Journey_Mode#Personal_Power_Menu).

If the sprite replacement tool goes up before this it's fine, I'm just a little concerned I'll need to "find" every variation of torch I've already used in order to wipe them out, and that's assuming the "replace" option can be blank in order to do this. A simpler "remove all torches" option as a kind of hard reset on torches could be handy.

Thanks either way.

**Comment by RussDev7** [comment]
I believe a better alternative to this would be to add “air” or “empty” to the masking. This way you can search and replace to nothing using the brush.

**Comment by GregariousJB** [comment]
> I believe a better alternative to this would be to add “air” or “empty” to the masking. This way you can search and replace to nothing using the brush.

I agree that having an "empty" option to the mask would be useful, but I disagree that it would be a better alternative. My suggestion would take two clicks. Also, my suggestion does not need to select from a list of 20+ different types of torches while masking or find & replacing.

**Comment by RussDev7** [comment]
> > I believe a better alternative to this would be to add “air” or “empty” to the masking. This way you can search and replace to nothing using the brush.
> 
> I agree that having an "empty" option to the mask would be useful, but I disagree that it would be a better alternative. My suggestion would take two clicks. Also, my suggestion does not need to select from a list of 20+ different types of torches while masking or find & replacing.

I don't know about making 20+ sprites for them but rather a generic torch mode. This sounds like something that a "Remove all sprites" plugin could do for selecting the name of a sprite to remove.

**Comment by GregariousJB** [comment]
> > > I believe a better alternative to this would be to add “air” or “empty” to the masking. This way you can search and replace to nothing using the brush.
> > 
> > 
> > I agree that having an "empty" option to the mask would be useful, but I disagree that it would be a better alternative. My suggestion would take two clicks. Also, my suggestion does not need to select from a list of 20+ different types of torches while masking or find & replacing.
> 
> I don't know about making 20+ sprites for them but rather a generic torch mode. This sounds like something that a "Remove all sprites" plugin could do for selecting the name of a sprite to remove.

I can sort of get behind this, although the brush size only goes up to 200, so you're waving your hand around the editor on a large map to try to remove all the torches while wishing it only took two clicks. If it also worked for the Selection Tool that would be better. But still...

I feel like a "just get rid of all the torches so I can redo them" option is still tough to beat. Especially when it's so easy, even in pre-hardmode, to fly back through the world and place them all again. Less time in the editor. More time in-game.

Wouldn't a plugin like my suggestion be easy to implement and be comparatively bug-free for a long time as well? When compared to more robust mechanics like find & replace or sprite replacing, that is. Loop through all the sprites -> If(torch), delete. Done.

**Comment by RussDev7** [comment]
> > > > I believe a better alternative to this would be to add “air” or “empty” to the masking. This way you can search and replace to nothing using the brush.
> > > 
> > > 
> > > I agree that having an "empty" option to the mask would be useful, but I disagree that it would be a better alternative. My suggestion would take two clicks. Also, my suggestion does not need to select from a list of 20+ different types of torches while masking or find & replacing.
> > 
> > 
> > I don't know about making 20+ sprites for them but rather a generic torch mode. This sounds like something that a "Remove all sprites" plugin could do for selecting the name of a sprite to remove.
> 
> I can sort of get behind this, although the brush size only goes up to 200, so you're waving your hand around the editor on a large map to try to remove all the torches while wishing it only took two clicks. If it also worked for the Selection Tool that would be better. But still...
> 
> I feel like a "just get rid of all the torches so I can redo them" option is still tough to beat. Especially when it's so easy, even in pre-hardmode, to fly back through the world and place them all again. Less time in the editor. More time in-game.
> 
> Wouldn't a plugin like my suggestion be easy to implement and be comparatively bug-free for a long time as well? When compared to more robust mechanics like find & replace or sprite replacing, that is. Loop through all the sprites -> If(torch), delete. Done.

Super easy to implement but a lack of demand for it. We don't want to add clutter to the editor, some users already find it over whelming. If I was to implement this I would like to make it universal to allow for removing other sprites as well.

**Comment by GregariousJB** [comment]
I suspect the demand might be higher than realized with the recent re-implementation of torch luck, but I'll defer to your judgment. I look forward to whatever solution happens. Thank you for your time. This can be closed if you wish.
